### PR TITLE
Update public CI to match changes to private CI

### PR DESCRIPTION
**Pull request by krzysz00**
This'll unblock merging PR #512 and improve CI build times.

(It'll also allow us to "cheat" CI if there's a passing run that gets
obliterated by an unrelated merge - I trust everyone not to abuse this)

### CODE REVIEW

File: mlir/utils/jenkins/Jenkinsfile.downstream [hunk 1]
```diff
@@ -45,23 +37,33 @@ String dockerImage() {
...
+                    when {
+                        beforeAgent true;
+                        equals expected: true, actual: params.sharedLib;
+                    }
                     agent {
                         docker {
                             image dockerImage()
                             args dockerArgs()
-                            label "rocm"
+                            label "gfx906"
```
Comment (jerryyin) [inline mlir/utils/jenkins/Jenkinsfile.downstream]: I think only static library does tuning and needs to run on gfx906 only. We can still use gfx900 on shared library tests.

File: mlir/utils/jenkins/Jenkinsfile.downstream [hunk 2]
```diff
@@ -98,16 +104,21 @@ pipeline {
...
                                 installation: 'InSearchPath', workingDir: 'build'
                             }
                         }
                         stage("Build MIOpen with libMLIRMIOpen") {
                             steps {
                                 dir('MIOpen') {
-                                getAndBuildMIOpen('--prefix ./MIOpenDeps', '''-DMIOPEN_USE_MLIR=On
+                                git branch: 'develop', poll: false,\
+                                    url: 'https://github.com/ROCmSoftwarePlatform/MIOpen.git'
+                                sh 'sed -i "/ROCmSoftwarePlatform\\/llvm-project-mlir/d" ./requirements.txt'
```
Comment (jerryyin) [inline mlir/utils/jenkins/Jenkinsfile.downstream:116]: Nice catch, after all those times we are testing an old branch. However, it does seem to be a reminder to extract common logic between private/public CI into a common place.

File: mlir/utils/jenkins/Jenkinsfile.downstream (hunk 1 above)
Comment (krzysz00) [reply mlir/utils/jenkins/Jenkinsfile.downstream]: Ah, I didn't know `rocm` implied a GPU. I'll roll this change back

**Review by jerryyin — APPROVED** [review]
LGTM

**Pull request merged**